src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -136,28 +148,68 @@ public boolean canUseLocalCache() {
 
         return false;
     }
-
+        
     /**
-     * App needs to give permission to AccountManager to use broker.
+     * To verify if App gives permissions to AccountManager to use broker.
+     * 
+     * @throws DeveloperAuthenticationException
+     * @return boolean If the GET_ACCOUNTS, MANAGE_ACCOUNTS, USE_CREDENTIALS permissions are granted in the Manifest.xml
      */
     private boolean verifyManifestPermissions() {
```

Comment:
It clearly says "Note that the declaration of the pop method does not contain a throws clause. EmptyStackException is not a checked exception, so pop is not required to state that it might occur." If you don't want to let caller know the exception might be thrown, then it's fine.